public/app/features/explore/utils/links.ts

```diff
@@ -70,7 +49,7 @@ export const getFieldLinksForExplore = (options: {
   range: TimeRange;
   vars?: ScopedVars;
   dataFrame?: DataFrame;
-}): Array<LinkModel<Field>> => {
+}): ExploreFieldLinkModel[] => {
```

Comment:
In practice `ExploreFieldLinkModel` is returned only for internal links. I think it might be worth adding documentation for it (here or at the type definition) as the `variables` property will always be created for internal links (could be an empty object) and will be undefined for non-internal links. 

Maybe we could also document that this was introduced to support any missing APIs in `LinkModel` objects that are required for Correlations. If we ever move Correlations out of Explore this will be needed to move to base LinkModel.